contracts/external/oracles/JonesUSDCWithChainlinkAutomationPriceOracle.sol

```diff
@@ -0,0 +1,115 @@
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+
+pragma solidity ^0.8.9;
+
+import { ChainlinkAutomationPriceOracle } from "./ChainlinkAutomationPriceOracle.sol";
+import { JonesUSDCMathLib } from "../jones/jusdc/JonesUSDCMathLib.sol";
+
+import { IDolomitePriceOracle } from "../../protocol/interfaces/IDolomitePriceOracle.sol";
+import { IDolomiteStructs } from "../../protocol/interfaces/IDolomiteStructs.sol";
+import { IERC4626 } from "../interfaces/IERC4626.sol";
+import { IJonesUSDCRegistry } from "../interfaces/jones/IJonesUSDCRegistry.sol";
+
+import { Require } from "../../protocol/lib/Require.sol";
+
+/**
+ * @title   JonesUSDCWithChainlinkAutomationPriceOracle
+ * @author  Dolomite
+ *
+ * @notice  An implementation of ChainlinkAutomationPriceOracle that gets Jones DAO's jUSDC price in USD terms
+ * @notice  Uses Chainlink automation
+ */
+contract JonesUSDCWithChainlinkAutomationPriceOracle is ChainlinkAutomationPriceOracle {
+    using JonesUSDCMathLib for IJonesUSDCRegistry;
+
+    // ============================ Constants ============================
+
+    bytes32 private constant _FILE = "jUSDCWithChainlinkPriceOracle";
+    uint256 private constant _USDC_DECIMALS_DIFF = 12;
+    uint256 private constant _USDC_SCALE_DIFF = 10 ** _USDC_DECIMALS_DIFF;
+
+    // ============================ Public State Variables ============================
+
+    IJonesUSDCRegistry immutable public JONES_USDC_REGISTRY; // solhint-disable-line var-name-mixedcase
+    uint256 immutable public USDC_MARKET_ID; // solhint-disable-line var-name-mixedcase
+    address immutable public DJUSDC; // solhint-disable-line var-name-mixedcase
+
+    // ============================ Constructor ============================
+    constructor(
+        address _dolomiteMargin,
+        address _chainlinkRegistry,
+        address _jonesUSDCRegistry,
+        uint256 _usdcMarketId,
+        address _djUSDC
+    ) ChainlinkAutomationPriceOracle(_dolomiteMargin, _chainlinkRegistry) {
+        JONES_USDC_REGISTRY = IJonesUSDCRegistry(_jonesUSDCRegistry);
+        USDC_MARKET_ID = _usdcMarketId;
+        DJUSDC = _djUSDC;
+
+        _updateExchangeRateAndTimestamp();
+    }
+
+    function getPrice(
+        address _token
+    )
+    public
+    view
+    returns (IDolomiteStructs.MonetaryPrice memory) {
+        Require.that(
+            _token == DJUSDC,
+            _FILE,
+            "Invalid token",
+            _token
+        );
+        Require.that(
+            DOLOMITE_MARGIN().getMarketIsClosing(DOLOMITE_MARGIN().getMarketIdByTokenAddress(_token)),
+            _FILE,
+            "jUSDC cannot be borrowable"
+        );
+        Require.that(
+            latestTimestamp + HEARTBEAT + GRACE_PERIOD > block.timestamp,
+            _FILE,
+            "price expired"
+        );
```

Comment:
Extract this `Require` into an internal function in `ChainlinkAutomationPriceOracle` that can be reused